lib/improver/utilities/mathematical_operations.py

```diff
@@ -0,0 +1,228 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module to contain mathematical operations."""
+
+import numpy as np
+
+import iris
+
+from improver.utilities.cube_manipulation import sort_coord_in_cube
+
+
+class Integration(object):
+    """Perform integration along a chosen coordinate."""
+
+    def __init__(self, coord_name_to_integrate,
+                 start_point=None, end_point=None,
+                 direction_of_integration="negative"):
+        """
+        Initialise class.
+
+        Args:
+            coord_name_to_integrate (iris.cube.Cube):
+                Name of the coordinate to be integrated.
+            start_point (float or None):
+                Point at which to start the integration.
+                Default is None.
+            end_point (float or None):
+                Point at which to end the integration.
+                Default is None.
+            direction_of_integration (string):
+                Description of the direction in which to integrate.
+                Options are 'positive' or 'negative'.
+                'positive' corresponds to the values within the array
+                increasing as the array index increases.
+                'negative' corresponds to the values within the array
+                decreasing as the array index increases.
+        """
+        self.coord_name_to_integrate = coord_name_to_integrate
+        self.start_point = start_point
+        self.end_point = end_point
+        self.direction_of_integration = direction_of_integration
+        if self.direction_of_integration not in ["positive", "negative"]:
+            msg = ("The specified direction of integration should be either "
+                   "'positive' or 'negative'. {} was specified.".format(
+                       self.direction_of_integration))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<Integration: coord_name_to_integrate: {}, '
+                  'start_point: {}, end_point: {}, '
+                  'direction_of_integration: {}>'.format(
+                      self.coord_name_to_integrate, self.start_point,
+                      self.end_point, self.direction_of_integration))
+        return result
+
+    def ensure_monotonic_increase_in_chosen_direction(self, cube):
+        """Ensure that the chosen coordinate is monotonically increasing in
+        the specified direction.
+
+        Args:
+            cube (Iris.cube.Cube):
+                The cube containing the coordinate to check.
+
+        Returns:
+            cube (Iris.cube.Cube):
+                The cube containing a coordinate that is monotonically
+                increasing in the desired direction.
+
+        Raises:
+            ValueError: The chosen coordinate is not monotonic.
```

Comment:
Is this accompanied by a useful error message?

Reply:
This error isn't actually being raised, so I've removed this from the docstring.